distributed/scheduler.py

```diff
@@ -4828,13 +4828,14 @@ def set_duration_estimate(self, ts: TaskState, ws: WorkerState):
         estimate the task duration to be 2x current-runtime, otherwise we set it
         to be the average duration.
         """
+        exec_time: double = ws._executing.get(ts, 0)
```

Comment:
Is it true that if `ts` is not in `ws._executing`, this will return 0 ?

Reply:
The main point is that we don't want to enter the `if` below. `0` was just a sensible default. Though we could also consider `-1`, which we have used elsewhere (as that is also an obviously bogus value for time).

`duration` is guaranteed to be positive (though maybe there is a vanishingly small chance that it is `0`). In any event `exec_time > 2 * duration` would be `False` even if `exec_time` is `0`, which is what we are trying to accomplish here.

Reply:
Thanks @jakirkham